env/vars.go

```diff
@@ -0,0 +1,42 @@
+package env
+
+import "strings"
+
+type Vars struct {
+	vals       map[string]string
+	ignoreCase bool
+}
+
+func NewVars(vars map[string]string, ignoreCase bool) *Vars {
```

Comment:
@ekcasey We debated whether to inline `ignoreCase` to `runtime.GOOS == "windows"` and eliminate it from the constructor. On the plus side of the current implementation, it does make it easy to test as a unit, and not have to switch OS targets to fully test it. Thoughts or preferences?

Reply:
Also some background: the reason for the new `Vars` type is that case sensitivity of env vars is now an important distinction between OSes, and we wanted to eliminate potential bugs due to folks working with maps directly.

Reply:
I like the decision to pass in an explicit `ignoreCase` or `caseSensitive` but I am wondering if inject it into `NewLaunchEnv` and `NewBuildEnv` instead of hardcoding `runtime.GOOS == "windows"` in `varsFromEnviron`. It makes sense to me from a domain modeling perspective. The entire environment is either case sensitive or case insensitive.

Reply:
That makes sense. I can make that change. There is also one other place where the constructor is called, in `WithPlatform()`, where I don't like that we have to check the runtime. Instead, I'll just copy the original `ignoreCase` value (look for a comment in the PR). That should leave checking the runtime GOOS in only the two places you mentioned.